Recommend enabling IRSA for new clusters (#12976)
kubernetes/kops: docs/getting_started/aws.md

aws s3api put-bucket-versioning --bucket prefix-example-com-state-store  --versioning-configuration Status=Enabled
```

In order for ServiceAccounts to use external permissions (aka IAM Roles for ServiceAccounts), you also need a bucket for hosting the OIDC documents.

> While you can reuse the bucket above if you grant it a public ACL, we do recommend a separate bucket for these files.

Reviewer: I wouldn't suggest giving the state/secret store a public ACL.

```suggestion
While you could reuse the bucket above by carefully managing ACLs, we do recommend a separate bucket for these files as the state store typically contains secrets.
```

The ACL must be public so that the AWS STS service can access them.

```bash
aws s3api create-bucket \
    --bucket prefix-example-com-oidc-store \

>     --region us-east-1

Author: ```suggestion
    --region us-east-1 \
```

```bash
kops create cluster \
    --zones=us-west-2a \

>     --discovery-store=s3://prefix-example-com-oidc-store/${NAME}/discovery

Author: ```suggestion
    --discovery-store=s3://prefix-example-com-oidc-store/discovery \
```
